tests/kernel/mem_protect/demand_paging/src/main.c

```diff
@@ -74,12 +86,112 @@ void test_touch_anon_pages(void)
 	printk("Kernel handled %lu page faults\n", faults);
 }
 
+void test_z_mem_page_out(void)
+{
+	unsigned long faults;
+	int key, ret;
+
+	/* Lock IRQs to prevent other pagefaults from happening while we
+	 * are measuring stuff
+	 */
+	key = irq_lock();
+	faults = z_num_pagefaults_get();
+	ret = z_mem_page_out(arena, HALF_BYTES);
+	zassert_equal(ret, 0, "z_mem_page_out failed with %d", ret);
+
+	/* Write to the supposedly evicted region */
+	for (size_t i = 0; i < HALF_BYTES; i++) {
+		arena[i] = nums[i % 10];
+	}
+	faults = z_num_pagefaults_get() - faults;
+	irq_unlock(key);
+
+	zassert_equal(faults, HALF_PAGES,
+		      "unexpected num pagefaults expected %lu got %d",
+		      faults, HALF_PAGES);
```

Comment:
I think the arguments are reversed. Should be "expected `HALF_PAGES` got `faults`".